erizo/src/erizo/stats/StatNode.cpp

```diff
@@ -0,0 +1,106 @@
+#include "stats/StatNode.h"
+
+#include <sstream>
+#include <string>
+#include <memory>
+#include <map>
+#include <iostream>
+
+#include "lib/ClockUtils.h"
+
+namespace erizo {
+
+typedef std::map<std::string, std::shared_ptr<StatNode>> NodeMap;
+
+StatNode& StatNode::operator[](std::string key) {
+  if (!hasChild(key)) {
+    node_map_[key] = std::make_shared<StatNode>();
+  }
+  return *node_map_[key];
+}
+
+std::string StatNode::toString() {
+  std::ostringstream text;
+  text << "{";
+  for (NodeMap::iterator node_iterator = node_map_.begin(); node_iterator != node_map_.end();) {
+    const std::string &name = node_iterator->first;
+    std::shared_ptr<StatNode> &node = node_iterator->second;
+    text << "\"" << name << "\":" << node->toString();
+    if (++node_iterator != node_map_.end()) {
+      text << ",";
+    }
+  }
+  text << "}";
+  return text.str();
+}
+
+StatNode& StringStat::operator=(std::string text) {
+  text_ = text;
+  return *this;
+}
+
+std::string StringStat::toString() {
+  std::ostringstream text;
+  text << "\"" << text_ << "\"";
+  return text.str();
+}
+
+StatNode& CumulativeStat::operator=(uint64_t initial) {
+  total_ = initial;
+  return *this;
+}
+
+StatNode CumulativeStat::operator++(int value) {
+  CumulativeStat node{total_};
```

Comment:
It's just the prototype used for the postincrement operator. There's more info here: http://en.cppreference.com/w/cpp/language/operator_incdec